### DISCUSSION TITLE
How to create control with blurry/acrylic background

### DISCUSSION THREAD

**Discussion by viktorspacil** [AvaloniaUI/Avalonia#16985] [Q&A]
Hi,
I am trying to create a control which allows me to use blurry/acrylic background (not only on the Window level which is supported by Avalonia by default). I based my solution on [this](https://gist.github.com/kekekeks/ac06098a74fe87d49a9ff9ea37fa67bc), but I am encountering glitches, for example when the button is hovered. Repro app is attached ([SampleApp.zip](https://github.com/user-attachments/files/16958771/SampleApp.zip)).

First render is ok:
<img width="632" alt="correct" src="https://github.com/user-attachments/assets/bed5fba2-f1b3-4a56-b71b-57b8bbe4ec71">

When I hover the boxed button or click the bottom button then the glitches appear:
<img width="621" alt="incorrect" src="https://github.com/user-attachments/assets/0098b799-3a6c-4252-bf3f-cb20c9ee9fd2">

The solution is based around skia.SkSurface.Snapshot() which helps to capture the control background and create the blur/acrylic effect. When the boxed button is hovered then the captured snapshot for the background is missing the spot where the button will be rendered which cause the glitch:
<img width="592" alt="blank spot under button" src="https://github.com/user-attachments/assets/e687af16-cc82-41d2-98cf-62cbd0ef6890">

More the same is happening when the bottom button is clicked:
<img width="583" alt="part of last panel missing" src="https://github.com/user-attachments/assets/72e8cfa9-d033-4b34-aa5a-1cfcd97dbc39">

It seems like some problem with the layering, could somebody help me with this? Thank you.

**Comment by maxkatz6** [comment]
Avalonia two `IEffect` implementations: `blur()` and `drop-shadow()`. You don't need to use custom render to get these effects.
Like:
```xaml
<Border Effect="blur(10)">
   <Button Content="Hi" />
</Border>
```

**Reply by maxkatz6** [reply]
> I am trying to create a control which allows me to use blurry/acrylic background (not only on the Window level which is supported by Avalonia by default)

Note, it's conceptually different.
Windows acrylic/mica is a host-backdrop blur algorithms, which is also supported on macOS with different implementation.

**Reply by viktorspacil** [reply]
Hi,
I am aware of blur() and drop-shadow() but these effects does not give me the result which I want (see the screenshots). Maybe I am using the effects wrong but I did not figure out the way to achieve the look like with using the custom rendering.

**Reply by viktorspacil** [reply]
Any ideas?
